src/instructlab/model/llmaaj.py

```diff
@@ -0,0 +1,164 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# pylint: disable=ungrouped-imports
+# Standard
+from datetime import datetime
+import enum
+import logging
+import pathlib
+import os
+
+# Third Party
+import httpx
+from openai import OpenAI, OpenAIError
+import pandas as pd
+from ragas.callbacks import ChainRun
+from ragas.dataset_schema import EvaluationDataset, EvaluationResult
+
+# First Party
+from instructlab.eval.ragas import ModelConfig, RagasEvaluator, RunConfig, Sample
+from instructlab.eval.mt_bench_common import get_openai_client
+from instructlab import client_utils
+
+# Local
+from ..utils import is_model_gguf, is_model_safetensors
+from ..client_utils import http_client, ClientException
+from .evaluate import launch_server, get_model_name
+from ..configuration import DEFAULTS
+
+logger = logging.getLogger(__name__)
+
+class IOFileType(enum.Enum):
+    CSV: str = "csv"
+    JSONL: str = "jsonl"
+    XLSX: str = "xlsx"
+    QNA_YAML: str = "qna.yaml"
+
+def model_is_local(model: str) -> bool:
+    if os.path.exists(model):
```

Comment:
Since the majority of your complex logic exists in a single condition, you can simplify this function by checking the opposite condition here and returning early. That way, the rest of your function can be on a single indent level:

```py
def is_local_model(model):
  if not os.path.exists(model):
    return False

  # continue here at the top-level indent
  if is_safetensors(model):
     ...
```